src/lib/migration-chunks/validation/editor-layout.ts

```diff
@@ -0,0 +1,163 @@
+  toBeCreatedFieldGroups: Set<string>
+}
+
+interface EditorLayoutValidation {
+  validate (intent: Intent, context: ValidationContext): string | string[]
+}
+
+class DuplicateCreate implements EditorLayoutValidation {
+  validate (intent: Intent, context: ValidationContext) {
+    if (!intent.isFieldGroupCreate()) {
+      return
+    }
+
+    if (!context.createdFieldGroups.has(getScopedFieldGroupId(intent))) {
+      return
+    }
+
+    return editorLayoutErrors.createFieldGroup.FIELD_GROUP_ALREADY_CREATED(
+      intent.getFieldGroupId(),
+      intent.getContentTypeId()
+    )
+  }
+}
+
+class AlreadyExistingCreates implements EditorLayoutValidation {
+  validate (intent: Intent, context: ValidationContext) {
+    if (!intent.isFieldGroupCreate()) {
+      return
+    }
+
+    if (!context.remoteFieldGroups.has(getScopedFieldGroupId(intent))) {
+      return
+    }
+
+    return editorLayoutErrors.createFieldGroup.FIELD_GROUP_ALREADY_EXISTS(
+      intent.getFieldGroupId(),
+      intent.getContentTypeId()
+    )
+  }
+}
+
+class NonExistingDeletes implements EditorLayoutValidation {
+  validate (intent: Intent, context: ValidationContext) {
+    if (!intent.isFieldGroupDelete()) {
+      return
+    }
+
+    const fieldGroupId = getScopedFieldGroupId(intent)
+
+    if (context.remoteFieldGroups.has(fieldGroupId) || context.deletedFieldGroups.has(fieldGroupId)) {
+      return
+    }
+
+    return editorLayoutErrors.deleteFieldGroup.FIELD_GROUP_DOES_NOT_EXIST(
+      intent.getFieldGroupId(),
+      intent.getContentTypeId()
+    )
+  }
+}
+
+class DuplicateDeletes implements EditorLayoutValidation {
+  validate (intent: Intent, context: ValidationContext) {
+    if (!intent.isFieldGroupDelete()) {
+      return
+    }
+
+    const fieldGroupId = getScopedFieldGroupId(intent)
+
+    if (!context.deletedFieldGroups.has(fieldGroupId)) {
+      return
+    }
+
+    return editorLayoutErrors.deleteFieldGroup.FIELD_GROUP_ALREADY_DELETED(
+      intent.getFieldGroupId(),
+      intent.getContentTypeId()
+    )
+  }
+}
+
+const checks: EditorLayoutValidation[] = [
+  new DuplicateCreate(),
+  new AlreadyExistingCreates(),
+  new NonExistingDeletes(),
+  new DuplicateDeletes()
+]
+
+function getScopedFieldGroupId (intent: Intent) {
+  return `${intent.getContentTypeId()}.${intent.getFieldGroupId()}`
+}
+
+export default function (intents: Intent[], editorInterfaces: Map<string, EditorInterfaces>): InvalidActionError[] {
+  let remote = []
+  editorInterfaces.forEach((editorInterfaces, ctId) => {
+    const editorLayout = editorInterfaces.getEditorLayout()
+    if (editorLayout) {
+      remote = remote.concat(collectFieldGroupIds(editorLayout).map(id => `${ctId}.${id}`))
+    }
+  })
+  const toBeCreated = intents.filter((intent) => intent.isFieldGroupCreate()).map(getScopedFieldGroupId)
+
+  let context = {
+    remoteFieldGroups: new Set(remote), // all currently (in the current iteration step) existing field groups
```

Comment:
I think those descriptions would be better kept at the interface definition above, so we get them as hints in editors whenever we hover over them.

Plus `let context: ValidationContext` maybe?